rtCamp/nginx-helper
# Sync Develop branch with Master

## Description
The aim of this PR is to sync the develop branch with master branch.

## Testing
The code has been tested locally and is working as expected.

To Test the functionality - 
1. Switch to the `develop` branch.
2. Ensure that all the basic functionality such as post,page,comment,feed are getting cached properly and purged when they are modified.
3. Ensure that purge all button works as intended.
4. Test custom URL's are getting purged.

## PR's included
- https://github.com/rtCamp/nginx-helper/pull/294
- https://github.com/rtCamp/nginx-helper/pull/334
- https://github.com/rtCamp/nginx-helper/pull/336
- https://github.com/rtCamp/nginx-helper/pull/347
- https://github.com/rtCamp/nginx-helper/pull/355
